core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -984,6 +986,14 @@ private void doFinish() {
     }
 
     /**
+     * Indicated whether this operation should be replicated to shadow replicas or not. If this method returns false the replication phase will be skipped.
```

Comment:
I think the comment is flipped - if the method returns true , shadow replicas will be skipped. I think we will better off making this a positive - executeOnShadowReplicas?  we can also make this method say "shouldExecuteReplication" and have the default implementation do `return IndexMetaData.isIndexUsingShadowReplicas(indexMetaData.settings()) == false`. That will make the if where the method is being used simpler.

Also, we need to adapt the unit testing of this class for this (I don't think we have a test now for shadow replicas)